File: HA10_5.c
//Accept N numbers from user and accept another number as NO, return frequency of NO from it
// N : 6
// NO : 66
// E : 85 66 3 66 93 88

#include<stdio.h>
#include<stdlib.h>

int Frequency(int Arr[], int iLength, int iNo)
{
    int iCnt = 0;
    int iCount = 0;

    for(iCnt = 0; iCnt < iLength; iCnt ++)
    {
        if(Arr[iCnt] == iNo)
        {
            iCount++;
        }
    }
    return iCount;
}

int main()
{
    int iSize = 0;
    int iCnt = 0;
    int iRet = 0;
    int iValue =0;
    int *ptr = NULL;

    printf("Enter the Number of Elements \n");
    scanf("%d", &iSize);

    printf("Enter the Number you want to check \n");
    scanf("%d", &iValue);

    ptr = (int*)malloc(iSize * sizeof(int));
    if(ptr == NULL)
    {
        printf("Unable to allocate memory \n");
        return -1;
    }

    printf("Enter the elements \n");
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        scanf("%d", &ptr[iCnt]);
    }

    printf("Elements of array are : \n");
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("%d\n", ptr[iCnt]);
    }

    iRet = Frequency(ptr, iSize, iValue);

    printf("No comes : %d\n",iRet ,"times");

    free(ptr);

    return 0;
}

File: HA10_5.py
def Frequency(arr, iNo):
    ECount = 0
    OCount = 0
    iCnt = 0

    for iCnt in arr:
        if iCnt  ==iNo:
            return True
    

def main():
    iSize = 0                                                            
    arr = []
    iCnt = 0
    iRet = 0
    iNo = 0

    iSize = int(input("Enter Number of Elements : "))

    iNo = int(input("Enter the Number you want to check : "))

    print("Enter Number of Elements : ")
    for iCnt in range(iSize):
        arr.append(int(input()))

    print("Elements in Array are : ")
    for iCnt in arr:
        print(iCnt)

    iRet = Frequency(arr, iNo)

    print("No is : ",iRet)

if __name__ =="__main__":
    main()